src/components/stack-chart/Canvas.js

```diff
@@ -59,6 +67,47 @@ const TEXT_OFFSET_TOP = 11;
 class StackChartCanvas extends React.PureComponent<Props> {
   _textMeasurement: null | TextMeasurement = null;
 
+  componentDidUpdate(prevProps) {
+    // We want to scroll the selection into view when this component
+    // is mounted, but using componentDidMount won't work here as the
+    // viewport will not have completed setting its size by
+    // then. Instead, look for when the viewport's isSizeSet prop
+    // changes to true.
+    const viewportDidMount =
+      !prevProps.viewport.isSizeSet && this.props.viewport.isSizeSet;
+
+    if (
+      viewportDidMount ||
+      this.props.scrollToSelectionGeneration >
+        prevProps.scrollToSelectionGeneration
+    ) {
+      this._scrollSelectionIntoView();
+    }
+  }
+
+  _scrollSelectionIntoView = () => {
+    const {
+      selectedCallNodeIndex,
+      callNodeInfo: { callNodeTable },
+    } = this.props;
+
+    if (selectedCallNodeIndex === null) {
+      return;
+    }
+
+    const depth = callNodeTable.depth[selectedCallNodeIndex];
```

Comment:
Is it is safe to assume that the depth of the call node is the same as the depth of the stack frame?

Reply:
Yes! That will always be true. The call nodes and stacks share the same function paths, but the call nodes represent aggregate frames at that function path, while each stack has a unique frame.